docs/backend/guides/service-objects/set-up-service-subpackage-and-interface.md

```diff
@@ -0,0 +1,214 @@
+---
+sidebar_position: 4
+---
+# Set Up Service Subpackage and Interface
+
+Since we're going to be creating data service objects, we'll be following the guidelines in
+[Structure - Data/Utility High Level Structure](/docs/backend/guides/service-objects/structure#datautility-service-objects-high-level-structure).
+
+For the purposes of this walk-through, that means creating a subpackage in the `services` package called `pet` and a 
+corresponding interface file called `pet.go`, like this:
+
+```text {5,7}
+mymove/
+├── pkg/
+│   ├── services/
+│   │   ├── ...
+│   │   ├── pet/                       <- our new subpackage
+│   │   ├── ...
+│   │   ├── pet.go                     <- our new interface in the services package
+│   │   ├── ...
+```
+
+## Service Interface
+
+The `services/pet.go` file will define the interface for our service objects. As noted in the 
+[structure](/docs/backend/guides/service-objects/structure] page, this file will only have the interface, while the 
+implementation logic will live in the subpackage. This means that the interface's functions that we define here will 
+have to match the receiver functions for our service object structs we create later in the implementation files.
+
+:::tip Interfaces vs Structs
+If you are new to Go and are still a little wobbly on the concept of "interfaces" vs "structs" remember:
+
+- **Interface** types define _functions_. They are concerned with _verbs_.
+- **Struct** types define _objects_. They are concerned with _nouns_.
+:::
+
+Let's start off by defining the bare interfaces for a `PetCreator` and a `PetUpdater`:
+
+```go title="pkg/services/pet.go"
+package services
+
+// PetCreator Interface for the service object that creates a pet
+type PetCreator interface {
+}
+
+// PetUpdater Interface for the service object that updates a pet
+type PetUpdater interface {
+}
+```
+
+Following our naming conventions stated in
+[Structure - Service Object Naming](/docs/backend/guides/service-objects/structure#service-object-naming), we'll 
+define the interface functions like so:
+
+```go title="pkg/services/pet.go"
+package services
+
+// PetCreator Interface for the service object that creates a pet
+type PetCreator interface {
+	CreatePet() (
```

Comment:
missing closing paren
```suggestion
	CreatePet() ()
```

Reply:
Ah, nice catch, thanks, I had copy/pasted a more complete version and then trimmed it down to be barebones, but I guess I trimmed too much ✂️